tool/gravity/cli/clusterupdate.go

```diff
@@ -392,3 +399,28 @@ Please use the gravity binary from the upgrade installer tarball to execute the
 
 	return nil
 }
+
+func selinuxBootstrap() error {
+	status, err := queryClusterStatus()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	selinuxEnabled := status.Cluster != nil && status.Cluster.SELinux
+	if !selinuxEnabled {
+		// Nothing to do
+		return nil
+	}
+	return selinux.Bootstrap(selinux.BootstrapConfig{})
+}
+
+func queryClusterStatus() (status *libstatus.Status, err error) {
+	out, err := exec.Command("gravity", "status", "--output=json").CombinedOutput()
```

Comment:
Why not use the API? Or localenv.ClusterOperation - seems like you only need the cluster above?

Reply:
Not really follow about the `localenv.ClusterOperation` - is this a new API?
The reason this does not use any APIs - the use of an API would require connecting to a port which the current executable might not be capable of - hence, it uses the existing binary (which should be correctly labeled on an installed cluster with SELinux previously enabled) to learn whether it should bootstrap.
If you have a better idea how to work around this chicken and egg problem - I'd be more than grateful.